### DISCUSSION TITLE
Explicitly access the intermediate layer features of the image encoder during inference

### DISCUSSION THREAD

**Discussion by JanineCHEN** [mlfoundations/open_clip#513] [Q&A]
Thank you for the great work!

I wonder if there is a fairly convenient way to explicitly access the intermediate layer features of the image encoder during inference? If this makes sense.

I have searched and couldn't find anything related to this in the prior issues (Maybe I have missed something, in this case, please kindly direct me). Thx so much!

**Comment by mitchellnw** [comment]
Not sure exactly what your use case is but if you put a print here https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/transformer.py#L244 it will print the intermediate features if your using a ViT base model.

**Reply by JanineCHEN** [reply]
Hi @mitchellnw , thank you for getting back to me. I'm trying to do a downstream image representation learning task and would like to leverage the representations learned by clip to the utmost by using the intermediate layer features. What I have in mind is to **access the intermediate layer features of the clip image encoder as frozen tensor variables**, such that I can concatenate them to the output tensors of another simpler network. I hope this could clarify my use case slightly better. Thx!
#513
